src/cpp/include/aiebu/aiebu_assembler.h

```diff
@@ -72,7 +124,7 @@ class aiebu_assembler
     buffer_type m_type;
     buffer_type m_output_type;
     class argtbl_impl;  // Forward declaration
-
+    file_artifact artifacts;
```

Comment:
This is the problem with not insulating public API classes.  We are now adding a new data member and have set ourself up for breaking backwards compatibility (ABI).  What if any existing binary is compiled against the version before artifacts was added?   It will not work because the artifacts member as seen by implementation, will be bogus (it will not exist) in the binary.  We cannot do this.